library/core/src/intrinsics/mod.rs

```diff
@@ -2102,6 +2102,48 @@ pub const fn saturating_add<T: Copy>(a: T, b: T) -> T;
 #[rustc_intrinsic]
 pub const fn saturating_sub<T: Copy>(a: T, b: T) -> T;
 
+/// Funnel Shift left, with wrapping on overflow.
+///
+/// Concatenates `a` and `b` (with `a` in the most significant half),
+/// creating an integer twice as wide. Then shift this vector left
+/// by `shift`, shifting in zeros, and extract the most significant
+/// half. If `a` and `b` are the same, this is equivalent to a rotate
+/// left operation.
```

Comment:
Please explicitly specify what happens when the shift is larger than the size of a / larger than the size of the concatenation. Does it still fill in zeroes, i.e. it ends up returning an all-0 value for the 2nd case?

Reply:
yeah, the 

> with wrapping on overflow

bit isn't clear enough I think. For funnel shifts the shift amount is modulo'd by the total number of bits in the type. so, a funnel shift on a `u8` by `12` is equivalent to one by `12 % u8::BITS`.